From ec3675d2b247aa641787d73d76d831b1f44069b9 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 27 Mar 2013 11:13:22 +0800
Subject: [PATCH 181/602] ENGR00256086 BT: modify configs for bt enable

1.add firmware items in PRODUCT_PACKAGES and trim some formats in imx*.mk.
2.restore bt support in require_hardware.xml.
3.unite some BT configs.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 imx53_smd/BoardConfig.mk                |  1 -
 imx5x/imx5x.mk                          | 19 +++++++++++--------
 imx6/etc/ueventd.freescale.rc           |  1 +
 imx6/imx6.mk                            | 33 +++++++++++++++++++--------------
 sabresd_6dq/BoardConfig.mk              |  3 ++-
 sabresd_6dq/bluetooth/bdroid_buildcfg.h | 33 +++++++++++++++++++++++++++++++++
 sabresd_6dq/required_hardware.xml       |  2 +-
 7 files changed, 67 insertions(+), 25 deletions(-)
 create mode 100644 sabresd_6dq/bluetooth/bdroid_buildcfg.h

diff --git a/imx53_smd/BoardConfig.mk b/imx53_smd/BoardConfig.mk
index 3c3e15a..b114b15 100755
--- a/imx53_smd/BoardConfig.mk
+++ b/imx53_smd/BoardConfig.mk
@@ -65,7 +65,6 @@ TARGET_USERIMAGES_USE_EXT4 := true
 
 # atheros 3k BT
 BOARD_USE_AR3K_BLUETOOTH := true
-BOARD_HAVE_BLUETOOTH_QCOM_ATH3K := true
 BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR := device/fsl/imx53_smd/bluetooth
 
 USE_OPENGL_RENDERER := true
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index 3f311ae..6a91948 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -24,7 +24,7 @@ PRODUCT_PACKAGES += \
 	librs_jni				\
 	pppd					\
 	chat					\
-        setup_fs                                \
+	setup_fs				\
 	ip-up-vpn				\
 	ip-up-ppp0				\
 	ip-down-ppp0				\
@@ -57,6 +57,7 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
 	audio.primary.imx5x			\
 	audio.usb.default			\
+	audio.a2dp.default			\
 	tinyplay				\
 	tinycap					\
 	tinymix					\
@@ -71,20 +72,22 @@ PRODUCT_PACKAGES += \
 	lights.imx5x				\
 	gralloc.imx5x				\
 	copybit.imx5x				\
-	hwcomposer.imx5x            		\
-	camera.imx5x            		\
+	hwcomposer.imx5x				\
+	camera.imx5x					\
 	power.imx5x				\
 	libbt-vendor				\
 	bt_vendor.conf				\
-        audio.r_submix.default                  \
+	audio.r_submix.default			\
 	magd
 
 # Bluetooth firmware files.
 PRODUCT_PACKAGES += \
-	ar3kbdaddr.pst				\
-	PS_ASIC.pst				\
-	RamPatch.txt				\
-	audio.a2dp.default			\
+	ar3kbdaddr_ar3001			\
+	PS_ASIC_ar3001				\
+	RamPatch_ar3001				\
+	ar3kbdaddr_ar3002			\
+	PS_ASIC_ar3002				\
+	RamPatch_ar3002
 
 # Freescale VPU firmware files.
 PRODUCT_PACKAGES += \
diff --git a/imx6/etc/ueventd.freescale.rc b/imx6/etc/ueventd.freescale.rc
index f8345dc..e667d04 100644
--- a/imx6/etc/ueventd.freescale.rc
+++ b/imx6/etc/ueventd.freescale.rc
@@ -1,4 +1,5 @@
 /dev/pmem_gpu             0660   system     graphics
+/dev/ttymxc4              0660   bluetooth  bluetooth
 /dev/snd/*                0664   system     audio
 /dev/ttyUSB*              0640   radio      radio
 /dev/ttyACM*              0640   radio      radio
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 2395edd..50c505e 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -15,7 +15,7 @@ PRODUCT_PACKAGES += \
 	Gallery2				\
 	Gallery		    			\
 	SoundRecorder				\
-        Camera                                  \
+	Camera					\
 	Email					\
 	FSLOta					\
 	VideoEditor				\
@@ -66,6 +66,7 @@ PRODUCT_PACKAGES += \
 	audio.primary.imx6			\
 	audio_policy.conf			\
 	tinyplay				\
+	audio.a2dp.default			\
 	tinycap					\
 	tinymix					\
 	libsrec_jni				\
@@ -81,17 +82,21 @@ PRODUCT_PACKAGES += \
 	lights.imx6				\
 	gralloc.imx6				\
 	copybit.imx6				\
-	hwcomposer.imx6             		\
-	camera.imx6                 		\
+	hwcomposer.imx6				\
+	camera.imx6				\
 	power.imx6				\
-        audio.r_submix.default                  \
+	audio.r_submix.default			\
+	libbt-vendor				\
 	magd
 
 # Bluetooth firmware files.
 PRODUCT_PACKAGES += \
-	ar3kdbaddr.pst				\
-	PS_ASIC.pst				\
-	RamPatch.txt
+	ar3kbdaddr_ar3001			\
+	PS_ASIC_ar3001				\
+	RamPatch_ar3001				\
+	ar3kbdaddr_ar3002			\
+	PS_ASIC_ar3002				\
+	RamPatch_ar3002				\
 
 # Freescale VPU firmware files.
 PRODUCT_PACKAGES += \
@@ -101,13 +106,13 @@ PRODUCT_PACKAGES += \
 
 # Atheros wifi firmwre files.
 PRODUCT_PACKAGES += \
-	fw-3                    		\
-	bdata                   		\
-	athtcmd_ram             		\
-	nullTestFlow            		\
-	cfg80211.ko             		\
-	compat.ko               		\
-	ath6kl_sdio.ko          		\
+	fw-3					\
+	bdata					\
+	athtcmd_ram				\
+	nullTestFlow				\
+	cfg80211.ko				\
+	compat.ko				\
+	ath6kl_sdio.ko				\
 	check_wifi_mac.sh
 
 # Atheros wifi tool
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 15f693a..22bdec8 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -79,7 +79,7 @@ $(error "TARGET_USERIMAGES_USE_UBIFS and TARGET_USERIMAGES_USE_EXT4 config open
 endif
 endif
 
-BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init video=mxcfb0:dev=ldb,bpp=32 video=mxcfb1:off video=mxcfb2:off fbmem=10M fb0base=0x27b00000 vmalloc=400M androidboot.console=ttymxc0 androidboot.hardware=freescale
+BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init video=mxcfb0:dev=ldb,bpp=32 video=mxcfb1:off video=mxcfb2:off fbmem=10M fb0base=0x27b00000 vmalloc=400M androidboot.console=ttymxc0 androidboot.hardware=freescale bluetooth
 
 ifeq ($(TARGET_USERIMAGES_USE_UBIFS),true)
 #UBI boot command line.
@@ -89,6 +89,7 @@ endif
 
 # atheros 3k BT
 BOARD_USE_AR3K_BLUETOOTH := true
+BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR := device/fsl/sabresd_6dq/bluetooth
 
 USE_ION_ALLOCATOR := false
 USE_GPU_ALLOCATOR := true
diff --git a/sabresd_6dq/bluetooth/bdroid_buildcfg.h b/sabresd_6dq/bluetooth/bdroid_buildcfg.h
new file mode 100644
index 0000000..20757ee
--- /dev/null
+++ b/sabresd_6dq/bluetooth/bdroid_buildcfg.h
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2012 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Copyright (C) 2013 Freescale Semiconductor, Inc. */
+
+#ifndef _BDROID_BUILDCFG_H
+#define _BDROID_BUILDCFG_H
+
+#define BTM_DEF_LOCAL_NAME "iMX6"
+
+// Networking, Capturing, Object Transfer
+// MAJOR CLASS: COMPUTER
+// MINOR CLASS: PALM SIZE PC/PDA
+#define BTA_DM_COD {0x1A, 0x01, 0x14}
+
+#define BTIF_HF_SERVICES (BTA_HSP_SERVICE_MASK)
+#define BTIF_HF_SERVICE_NAMES  { BTIF_HSAG_SERVICE_NAME }
+#define PAN_NAP_DISABLED TRUE
+
+#endif
diff --git a/sabresd_6dq/required_hardware.xml b/sabresd_6dq/required_hardware.xml
index 2cde2ee..a6e0dcb 100644
--- a/sabresd_6dq/required_hardware.xml
+++ b/sabresd_6dq/required_hardware.xml
@@ -26,7 +26,7 @@
     <feature name="android.hardware.location.network" />
     <feature name="android.hardware.sensor.accelerometer" />
     <feature name="android.hardware.sensor.compass" />
-    <!-- <feature name="android.hardware.bluetooth" /> -->
+    <feature name="android.hardware.bluetooth" />
     <feature name="android.hardware.touchscreen" />
     <feature name="android.hardware.touchscreen.multitouch" />
     <feature name="android.hardware.touchscreen.multitouch.distinct" />
-- 
1.8.0

